samples/wifi/provisioning/implementation_details.rst

```diff
@@ -0,0 +1,268 @@
+If the device meets an error when getting value, the status of response message is INTERNAL_ERROR, and the whole device_status shall be discarded by the configurator.
+
+START_SCAN procedure
+--------------------
+
+When request with Op Code of START_SCAN is received, the device shall send a command of start scan to Wi-Fi subsystem, and depending on the result, a response message is sent to configurator.
+
+If the scan_params in request message is present, it is used to control the scan.
+If the operation is successfully dispatched, the device shall send response message with status of STATUS_SUCCESS.
+If invalid argument is provided, the status shall be STATUS_INVALID_ARGUMENT.
+If internal error occurs, the status shall be STATUS_INTERNAL_ERROR.
+
+If the operation is successful, the device shall send the scan result.
+The message shall be in type of Result defined in result.proto and encoded.
+The information of access point shall be in the scan_record field.
+In each sending function call, it should put only one scan result.
+
+STOP_SCAN procedure
+-------------------
+
+When request with Op Code of STOP_SCAN is received, the device shall send a command of stop scan to Wi-Fi subsystem, and depending on the result, a response message is sent to configurator.
+
+If the operation is successful, the device shall send response message with status of STATUS_SUCCESS.
+If internal error occurs, the status shall be STATUS_INTERNAL_ERROR.
+
+SET_CONFIG procedure
+--------------------
+
+When request with Op Code of SET_CONFIG is received, the device shall send a command of connecting to the Access Point specified in the config field to Wi-Fi subsystem, and depending on the result, a response message is sent to configurator.
+
+If the operation is successfully dispatched, the device shall send response message with status of STATUS_SUCCESS.
+If invalid argument is provided, the status shall be STATUS_INVALID_ARGUMENT.
+If internal error occurs, the status shall be STATUS_INTERNAL_ERROR.
+
+If the operation is successful, the device shall send state update to configurator.
+The message shall be in type of Result defined in result.proto and encoded.
+The updated state of Wi-Fi should be in state field. If the state is CONNECTION_FAILED, the device may fill reason field, whose value depends on the failure reason.
+If the state is CONNECTED, the device shall store the credentials in NVM.
+
+FORGET_CONFIG procedure
+-----------------------
+
+When request with Op Code of FORGET_CONFIG is received, the device shall remove stored credentials from NVM, and depending on the result, a response message is sent to configurator.
+
+If the operation is successful, the device shall send response message with status of STATUS_SUCCESS.
+And the device shall disconnect from access point if its state is connected.
+If internal error occurs, the status shall be STATUS_INTERNAL_ERROR.
+
+You may find the details of definition in .proto files under path of ./common/proto.
```

Comment:
Rephrase and move to the beginning of the section, as now it looks like this only applies to FORGET_CONFIG.

See `./common/proto` files for more information on definitions.